### ISSUE TITLE
Core: enable cutting geometry (e.g. circular or inserting defects)

### ISSUE THREAD

**Issue by GPMueller** [spirit-code/spirit#104]
This could be achieved through a mask of 0 and 1 for spins which are cut out and counted respectively.

One question is: how to signify this for the visualisation? Set the spin length to 0 (might conflict when re-normalizing, unless the mask is applied again after)?

**Comment by GPMueller** [comment]
Requires: #102

**Comment by GPMueller** [comment]
Currently this can be achieved two ways:
- set certain atoms as vacancies via atom types (see #331 for doing this interactively in the UI)
- take a single big unit cell, corresponding to the desired geometry, as input

The visualisation sets the spin lengths separately from the core, so that spins with `atom_type=-1` are simply set to length 0 to make them invisible.

**Next Step**: set up a few utilities which enable cylindrical, spherical etc. cutting by setting atom types (see also the filter function used by the configuration setters).
Also consider what it would mean to actually change the geometry in this way at runtime, i.e. actually changing the number of spins.

**Comment by GPMueller** [comment]
Implemented with 5d837b676e960f5a6fd229097d5d721586a1438d, merged into `develop` with eb841c5bd39941d4c437895845969c00ff224f10.
Finished with 9db0979a63ab28088918613daffe43d2a0329ac5.
